tree_segment.py

```diff
@@ -0,0 +1,434 @@
+import laspy
```

Comment:
Please add a file-level docstring: https://google.github.io/styleguide/pyguide.html#382-modules
e.g.
"""Segments point cloud data from a LAS file into individual trees.

This program reads lidar data from a LAS file, identifies clusters of points corresponding to individual trees, and outputs the resulting clusters in a csv file.

In order to identify trees, this algorithm first identifies vertical "stems" which go from the top of a tree to the ground. The remaining points are matched to nearby stems, resulting in one output cluster per stem which should correspond to a single tree.

Example Usage:
  python3 tree_segment.py --file_name sample_data/treeID_40645_merged.las
"""

Extra Braid Commentary:
All of this information is well covered in the README, the point here is more to include at least minimal documentation in the file itself (e.g. you could link to a section in the README). This helps people who encounter the file first, and lets you create new files without worrying about keeping the README in sync.

I think it would help to separate the tree finding algorithm from the file input/output operations, I'll add a top-level comment with more details.

Reply:
Docstring added. Having separate files would probably be good, but I'm going to keep it as-is to minimize disruption to my coauthor's workflow.